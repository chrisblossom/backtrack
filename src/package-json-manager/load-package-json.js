/* @flow */

import path from 'path';
import { existsSync } from 'fs';
import { rootPath } from '../config/paths';

import type { PackageJson } from '../types.js';

function loadPackageJson(): PackageJson {
    const file = path.resolve(rootPath, 'package.json');
    const exists = existsSync(file);

    if (exists === false) {
        throw new Error('package.json not found');
    }

    const load = require(file);
    return load;
}

export { loadPackageJson };
