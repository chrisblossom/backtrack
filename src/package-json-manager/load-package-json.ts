import path from 'path';
import { existsSync } from 'fs';
import { rootPath } from '../config/paths';

import { PackageJson } from '../types';

function loadPackageJson(): PackageJson {
    const file = path.resolve(rootPath, 'package.json');
    const exists = existsSync(file);

    if (exists === false) {
        throw new Error('package.json not found');
    }

    const load = require(file);
    return load;
}

export { loadPackageJson };
