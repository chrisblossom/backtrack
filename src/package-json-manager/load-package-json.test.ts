import path from 'path';
import type { loadPackageJson as loadPackageJsonType } from './load-package-json';

type LoadPackageJson = typeof loadPackageJsonType;
const loadPackageJson = async (): ReturnType<LoadPackageJson> =>
	require('./load-package-json').loadPackageJson();

describe('loadPackageJson', () => {
	const cwd = process.cwd();

	afterEach(() => {
		process.chdir(cwd);
	});

	test('loads stat file', async () => {
		const dir = path.resolve(__dirname, '__sandbox__/found-package-json/');
		process.chdir(dir);

		const result = await loadPackageJson();

		expect(result).toEqual({
			name: 'found_package_json',
			private: true,
		});
	});

	test('fail when package json does not exist', async () => {
		const dir = path.resolve(
			__dirname,
			'__sandbox__/missing-package-json/',
		);
		process.chdir(dir);

		let error;
		try {
			await loadPackageJson();
		} catch (e) {
			error = e;
		} finally {
			expect(error).toMatchInlineSnapshot(
				`[Error: package.json not found]`,
			);
		}
	});
});
