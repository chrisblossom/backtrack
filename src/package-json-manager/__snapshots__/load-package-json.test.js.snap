// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadPackageJson fail when package json does not exist 1`] = `[Error: package.json not found]`;

exports[`loadPackageJson loads stat file 1`] = `
Object {
  "name": "found_package_json",
  "private": true,
}
`;
